### ISSUE TITLE
Compilation on old-box Fedora 21

### ISSUE THREAD

**Issue by tathanhdinh** [unicorn-engine/unicorn#500]
Hello everyone and many thanks for an execellent emulator.
I have encountered a trivial problem (I think) when compiling `unicorn` (release version 0.9) on my box `fedora 21` (yes, it is old but I cannot upgrade it now because of some ABI compatibility problems with libraries of my current project). I have typed `./make.sh` with the edited `config.mk`

> UNICORN_DEBUG ?= no
> UNICORN_ARCHS ?= x86
> UNICORN_STATIC ?= yes
> UNICORN_SHARED ?= yes

then received some linkage errors:

```
  LINK    sample_x86
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_init'
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_settype'
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_destroy'
../libunicorn.so: error: undefined reference to 'pthread_mutex_trylock'
../libunicorn.so: error: undefined reference to 'sem_init'
../libunicorn.so: error: undefined reference to 'sem_destroy'
../libunicorn.so: error: undefined reference to 'sem_post'
../libunicorn.so: error: undefined reference to 'sem_trywait'
../libunicorn.so: error: undefined reference to 'sem_timedwait'
../libunicorn.so: error: undefined reference to 'sem_wait'
../libunicorn.so: error: undefined reference to 'pthread_sigmask'
../libunicorn.so: error: undefined reference to 'pthread_create'
../libunicorn.so: error: undefined reference to 'pthread_join'
../libunicorn.so: error: undefined reference to 'pthread_kill'
collect2: erreur: ld a retourné 1 code d'état d'exécution
Makefile:120: recipe for target 'sample_x86' failed
make[1]: *** [sample_x86] Error 1
make[1]: Leaving directory '/home/bigraph/Public/unicorn-0.9/samples'
Makefile:188: recipe for target 'all' failed
make: *** [all] Error 2
```
It is indeed trivial to fix by changing `LDFLAGS` in `sample/Makefile` into:

`LDFLAGS += -L$(LIBDIR) -l$(LIBNAME) -lpthread`

This is natural to me since `pthread` should be added, but I am not sure it is the same on others OS or versions of Linux.

**Comment by aquynh** [comment]
first of all, this is the 500th issue of Unicorn, congrats!

secondly, you dont need to modify `config.mk`, but just pass the settings on the commandline, like this:

    UNICORN_ARCHS=x86 UNICORN_SHARED=yes ./make.sh

finally, it seems you are using very old version of Unicorn, since the latest code from Github already includes pthread in LDFLAGS: https://github.com/unicorn-engine/unicorn/blob/master/samples/Makefile#L30. we actually use pthread everywhere, even for Windows where we simply cross-compile Unicorn using Msys.

you are strongly recommended to use the latest code from Github, which significantly improves since v0.9. be sure to always reinstall the bindings with the core to avoid any unwanted issues.

thanks.

**Comment by aquynh** [comment]
i missed the part you said you are using v0.9. it is a good idea to avoid that and use the latest code from Github instead.

**Comment by tathanhdinh** [comment]
Ah, many many thanks for your response. I have download the version 0.9 (package tar.gz) since it is recommended from the website of [Unicorn](http://www.unicorn-engine.org/download/). Indeed, the lastest source is compiled smoothly.
Thanks again for your prompt response.

**Comment by tathanhdinh** [comment]
This is not important, but could you please add an option for the installation path (instead of the default `/usr/local`)

**Comment by lunixbochs** [comment]
Seconded, I install manual packages on OS X into homebrew's Cellar so they can be linked/unlinked/versioned properly.

**Comment by aquynh** [comment]
have you tried with:

    $sudo PREFIX=/usr ./make.sh install

?

**Comment by lunixbochs** [comment]
https://github.com/unicorn-engine/unicorn/blob/master/make.sh#L54

**Comment by aquynh** [comment]
i think changing that to:

    PREFIX?=/usr/local

can fix this issue

**Comment by lunixbochs** [comment]
this is the middle of a shell script, not a makefile - the bash syntax for that is `PREFIX="${PREFIX-/usr/local}"`

**Comment by aquynh** [comment]
oh yes. send in a PR, please :-)

**Comment by aquynh** [comment]
merged the PR #502, so you can pass PREFIX at install time now, thanks.

**Comment by tathanhdinh** [comment]
Thanks a lot for your help.
